src/cata_tiles.cpp

```diff
@@ -290,61 +290,76 @@ static void get_tile_information( std::string config_path, std::string &json_pat
     }
 }
 
-static inline Uint8 average_pixel_color( const SDL_Color &color )
-{
-    return 85 * ( color.r + color.g + color.b ) >> 8; // 85/256 ~ 1/3
-}
+// A wrapper of SDL_Color
+struct pixel {
+    union {
+        SDL_Color color;
+        struct {
+            Uint8 r;
+            Uint8 g;
+            Uint8 b;
+            Uint8 a;
+        };
+        Uint32 color_data;
```

Comment:
This is a popular misconception about unions in C++ known as type-punning. Some compilers explicitly support it (for example, gcc in GNU extensions), it may be working in other compilers too, but I'd refrain from using it in portable code since it's undefined by the standard.